payment/payment.go

```diff
@@ -38,10 +38,8 @@ func FromPublicKey(pubkey *btcec.PublicKey, network *network.Network) *Payment {
 
 // FromPayment creates a Payment struct from a another Payment
 func FromPayment(payment *Payment) *Payment {
-	buf := payment.Script
-	hash := hash160(buf)
-	redeem := &Payment{payment.Network, payment.PublicKey, hash, nil, nil, nil}
-	return &Payment{nil, nil, nil, nil, redeem, nil}
+	redeem := &Payment{payment.Network, payment.PublicKey, payment.Hash, nil, nil, nil}
+	return &Payment{payment.Network, nil, payment.Hash, nil, redeem, nil}
```

Comment:
The `Hash` of the payment should be, as in the previous commit, the hash of the redeem script:
```suggestion
	return &Payment{payment.Network, nil, hash160(redeem.Script), nil, redeem, nil}
```

Reply:
Because of this, even before instantiating `redeem` we should check that `payment.Script != nil`, returning an error otherwise.